Portal/src/Datahub.Portal/Components/EnvironmentVariablesTable.razor

```diff
@@ -0,0 +1,168 @@
+            <MudTh>
+                <MudTableSortLabel SortBy="@(new Func<(string Key, string Value), string>(x => x.Key))">@Localizer["Key"]</MudTableSortLabel>
+            </MudTh>
+            <MudTh>
+                <MudTableSortLabel SortBy="@(new Func<(string Key, string Value), string>(x => x.Value))">@Localizer["Value"]</MudTableSortLabel>
+            </MudTh>
+            <MudTh/>
+        </HeaderContent>
+        <RowTemplate>
+            <MudTd DataLabel="@Localizer["Key"]">@context.Key</MudTd>
+            <MudTd DataLabel="@Localizer["Value"]">@context.Value</MudTd>
+            <MudTd>
+                <MudTooltip Text="@Localizer["Delete environment variable"]">
+                    <MudIconButton Size="Size.Small" Icon="@SidebarIcons.Delete" aria-label="@(Localizer["Delete"])"
+                                   OnClick="@(() => DeleteEnvironmentVariable(context, true))"/>
+                </MudTooltip>
+            </MudTd>
+        </RowTemplate>
+        <RowEditingTemplate>
+            <MudTd DataLabel="@Localizer["Key"]">
+                <MudTextField @bind-Value="@context.Key" Required/>
+            </MudTd>
+            <MudTd DataLabel="@Localizer["Value"]">
+                <code>
+                    <MudTextField @bind-Value="@context.Value" Required/>
+                </code>
+            </MudTd>
+            <MudTd/>
+        </RowEditingTemplate>
+        <EditButtonContent Context="button">
+            <MudIconButton Size="@Size.Small" Icon="@SidebarIcons.Edit" Class="pa-0" OnClick="@button.ButtonAction" Disabled="@button.ButtonDisabled"/>
+        </EditButtonContent>
+    </MudTable>
+
+    <MudButton Variant="Variant.Filled" OnClick="@AddNewEnvironmentVariable" StartIcon="@SidebarIcons.Plus">
+        @Localizer["Add environment variable"]
+    </MudButton>
+}
+else
+{
+    <MudTable Items="@envVars"
+              Dense="false"
+              Hover="false"
+              Outlined="true"
+              Class="mt-6 mb-12"
+              Filter="@(new Func<(string Key, string Value), bool>(FilterFunc))"
+              SortLabel="@Localizer["Sort by"]">
+
+        <ToolBarContent>
+            <MudText Typo="Typo.h3" Class="mt-6">
+                @Localizer["Environment variables"]
+            </MudText>
+            <MudSpacer/>
+            <MudTextField
+                @bind-Value="@_filterString"
+                Label="@Localizer["Filter"]"
+                Adornment="Adornment.End"
+                AdornmentIcon="@SidebarIcons.Search"
+                IconSize="Size.Small"
+                Class="mt-0"/>
+            <MudSpacer/>
+            <MudToggleIconButton @bind-Toggled="@_show" Icon="@SidebarIcons.Show" ToggledIcon="@SidebarIcons.Hide" Title="@Localizer["Show environment variable values"]" ToggledTitle="@Localizer["Hide environment variable values"]"/>
+        </ToolBarContent>
+
+        <ColGroup>
+            <col/>
+            <col/>
+            <col/>
+        </ColGroup>
```

Comment:
Was originally going to make a column for a "show value"/"hide value" button for each row to show/hide single rows instead of the full table but scrapped that halfway through. Will remove this, it is not needed.